src/components/Api.js

```diff
@@ -0,0 +1,86 @@
+export default class Api {
+  constructor(options) {
+    this._baseUrl = options.baseUrl;
+    this._headers = options.headers;
+  }
+
+  _checkResponse(res) {
+    if (res.ok) {
+      return res.json();
+    } else {
+    return Promise.reject(`Error: ${res.status}`);
+  }
+  }
+
+  async _request(url, options) {
```

Comment:
Needs correction: There is no need to mark the function as `async` here, since you don't use `await` statements inside